src/Swift.Runtime/src/Library/ISwiftObject.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Swift.Runtime;
+
+/// <summary>
+/// Represents a type that can be marshaled into Swift
+/// </summary>
+public interface ISwiftObject
+{
+    /// <summary>
+    /// Returns the TypeMetadata for this object
+    /// </summary>
+    /// <returns>A type metadata object for the type.</returns>
+    static abstract TypeMetadata GetTypeMetadata();
```

Comment:
This is in an interface, so contrary to everywhere else, visibility defaults to `public`, which is a bit confusing (especially if a default implementation is provided). Thus maybe make it explicit:

```suggestion
    public static abstract TypeMetadata GetTypeMetadata();
```